File: app/src/main/java/ro/scoalainformala/trips/Trip/TripDao.java
package ro.scoalainformala.trips.Trip;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface TripDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Trip trip);

    @Update
    void update(Trip trip);

    @Query("DELETE FROM trip_table")
    void deleteAll();

    @Query("SELECT * FROM trip_table ORDER BY title ASC")
    LiveData<List<Trip>> getAlphabetizedTrips();

    @Query("SELECT * FROM trip_table WHERE isFavourite='true'")
    LiveData<List<Trip>> getAllFavouriteTrips();
}

File: app/src/main/java/ro/scoalainformala/trips/Trip/TripDao.kt
package ro.scoalainformala.trips.Trip

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update

@Dao
interface TripDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(trip: Trip?)

    @Update
    fun update(trip: Trip?)

    @Query("DELETE FROM trip_table")
    fun deleteAll()

    @get:Query("SELECT * FROM trip_table ORDER BY title ASC")
    val alphabetizedTrips: LiveData<List<Trip?>?>?

    @get:Query("SELECT * FROM trip_table WHERE isFavourite= 1")
    val allFavouriteTrips: LiveData<List<Trip?>?>?
}
